Toggle switch children

alandardic opened the pull request:
Hi @lbelinsk,

I have added children property to ToggleSwitch + tests.

Please review 

Thanks

lbelinsk commented on packages/wix-ui-core/src/components/ToggleSwitch/ToggleSwitch.driver.ts:
```diff
@@ -24,6 +25,6 @@ export const toggleSwitchDriverFactory = ({element, componentInstance, eventTrig
         property: 'border-radius'
       });
-    }
+    },
```
remove comma

lbelinsk commented on packages/wix-ui-core/src/components/ToggleSwitch/ToggleSwitch.protractor.driver.ts:
```diff
@@ -2,5 +2,6 @@ export const toggleSwitchDriverFactory = component => ({
   click: () => component.click(),
   isChecked: () => component.$('input').isSelected(),
-  isDisabled: () => !component.$('input').isEnabled()
+  isDisabled: () => !component.$('input').isEnabled(),
```
Good work with the tests.
Now what we still need is docs.
Will call you guys after the review

lbelinsk commented on packages/wix-ui-core/stories/index.tsx:
```diff
@@ -19,7 +19,7 @@ storiesOf('Components', module)
   ))
   .add('ToggleSwitch', () => (
-    <ToggleSwitch/>
+    <ToggleSwitch dataHook="story-toggle-switch">Hello<i>?</i></ToggleSwitch>
```
why the question mark? hello is enough, no ? :)

alandardic commented:
Hi @lbelinsk,

I have fixed all the pr comments and added the comments you have asked (for Badge and ToggleSwitch drivers).

Please review and merge with master (if it's ready)

Thanks

lbelinsk commented on packages/wix-ui-core/src/components/Badge/Badge.driver.ts:
```diff
@@ -8,17 +8,29 @@ export const badgeDriverFactory = ({element, componentInstance, eventTrigger}) =
+    /** getType method - returns elements innerHTML */
     getContent: () => element.innerHTML,
+    /** getHeight method - returns elements height css property*/
     getHeight: () => domTestkit.getCssValue({className: 'badge', property: 'height'}),
```
Don't kill me, but I had an idea maybe to have another layer for the "style" methods.
```
return {
  exists: () => !!element,
  exists: () => !!element,
  getContent: () => element.innerHTML,
  styles: {
      getHeight: () => domTestkit.getCssValue(...,
      getPadding: () => domTestkit.getCssValue(...,
      ....
   }
}
```

lbelinsk commented:
Guys, so much mixing between pr's (Badge & Toggleswitch are mixed)

argshook commented on packages/wix-ui-core/src/components/ToggleSwitch/ToggleSwitch.e2e.ts:
```diff
@@ -0,0 +1,18 @@
+    const driver = toggleSwitchTestkitFactory({dataHook});
+
+    return waitForVisibilityOf(driver.element(), 'Cannot find ToggleSwitch')
+      .then(() => expect(driver.getTextContent()).toBe('Hello'));
```
e2e infrastructure ready === autotools ready
unfortunately above currently evaluates to `false`

to make that `true` autotools needs to:
1. better understand typescript
1. resolve components that are wrapped in HOCs (like `withClasses`)

alandardic replied:
Hi @lbelinsk,

So, currently I can't check toggle checked/ unchecked functionality - using e2e, right?

argshook commented on packages/wix-ui-core/src/components/Badge/Badge.driver.ts:
```diff
@@ -8,17 +8,29 @@ export const badgeDriverFactory = ({element, componentInstance, eventTrigger}) =
   return {
+    /** exists method - checks if element exists */
     exists: () => !!element,
+    /** getType method - returns elements type */
```
@alandardic hey, no need to label each, autodocs (when they work ;) ) will use actual method name. so just description is enough

lbelinsk commented on packages/wix-ui-core/src/components/ToggleSwitch/ToggleSwitch.protractor.driver.ts:
```diff
@@ -1,6 +1,12 @@
 export const toggleSwitchDriverFactory = component => ({
+  /** element method - returns the component element */
```
@argshook already commented about it here :)

lbelinsk commented on packages/wix-ui-core/src/components/ToggleSwitch/theme.ts:
```diff
@@ -1,35 +1,38 @@
+  toggleIconHeight: string;
+  toggleIconDisplay: string;
+
+  childrenColor: string;
```
childrenColor is too low level, and talks about implementation.
What ablit LabelColor?
